blockchain/utxobackend.go

```diff
@@ -15,55 +15,126 @@ import (
 	"github.com/decred/dcrd/blockchain/standalone/v2"
 	"github.com/decred/dcrd/chaincfg/chainhash"
 	"github.com/decred/dcrd/chaincfg/v3"
-	"github.com/decred/dcrd/database/v2"
 	"github.com/decred/dcrd/wire"
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/filter"
+	"github.com/syndtr/goleveldb/leveldb/opt"
+	"github.com/syndtr/goleveldb/leveldb/util"
 )
 
 const (
 	// currentUtxoDatabaseVersion indicates the current UTXO database version.
-	currentUtxoDatabaseVersion = 1
+	currentUtxoDatabaseVersion = 2
 
-	// currentUtxoSetVersion indicates the current UTXO set database version.
-	currentUtxoSetVersion = 3
+	// utxoDbDir is the directory for the UTXO database.
+	utxoDbDir = "utxodb"
 
-	// utxoDbName is the UTXO database name.
-	utxoDbName = "utxodb"
+	// utxoLdbName is the name of UTXO leveldb database.  It is itself within the
+	// utxoDbDir directory.
+	utxoLdbName = "metadata"
```

Comment:
There is not a reason other than it'll require a bit more work in the initialization and upgrade code, so I had punted and left it the same (we'd need to try to open the database from both locations in initialization and then for migration open both databases and move the data from one to another).  I'll take another look at it while I'm making the other updates and see if it's worth doing now.

Reply:
Ah, right, the utxo db was separated out using `database` which created it that way, so you'd have do more migration work to move it.

I'm not sure what you mean about needing to open the db in both locations in initialization though.  I would expect that the migration code would know it needs to move it based on the overall database version and the initialization code only opens the latest since it runs after any potential migrations.

Reply:
Yeah, exactly.  The initialization part I was referring to is that you need to open the database in order to read the overall database version.  If the database doesn't exist in the new location, we'd need to check the old location in order to know which version to start from (and then proceed with migrating).

Reply:
I decided to add a separate commit (`multi: Move UTXO database.`) to remove the need for the metadata directory.  I did it separately from the other migration for a couple of reasons:
- It is easier to follow versus adding more complexity into the existing migrations
- It is only going to run for those that have been running `dcrd` on master post-UTXO database separation (so for the vast majority of users that just use released versions, the code to move the database will never run, and it will just be created in the new location to begin with).

Reply:
I agree with your assessment.  It is certainly easier to follow as implemented and it is only going to be required by a very small number of people anyway.  I should also note for the transparency that we discussed this a bit offline and decided that it was easier to do the migration as it is now implemented instead of trying to move the file since that runs into various potential issues with things such as `os.Rename` not working across drives and network paths.